src/main/java/net/minecraftforge/fml/common/network/NetworkEventFiringHandler.java

```diff
@@ -49,13 +47,15 @@
     protected void channelRead0(ChannelHandlerContext ctx, FMLProxyPacket msg) throws Exception
     {
         eventChannel.fireRead(msg,ctx);
+        msg.payload().release();
```

Comment:
This will cause any mod trying to use the payload to break, because the payload is likely to be recycled. It should be released by the downstream system or the mod itself, explicitly (Event handler is not for the simple user).

What analysis caused you to conclude that there was a leak here? Your test case change below shows literally zero analysis or demonstration of expected vs unexpected.

Reply:
https://gist.github.com/bs2609/f91732ab507f81da538f58ab7a813cbe

Reply:
@Barteks2x, do you want to offer an opinion as you are more familiar with netty?

Reply:
We can have a "better" solution, that is releasing the buff with `finalize` method when the payload is finalized with bytecode engineering. (of course it sounds worse) I haven't seen any report on "networking's fucked" yet, so this one should be fine.